strawberry/federation/schema.py

```diff
@@ -68,12 +68,10 @@ def _extend_query_type(self):
 
         query_type = cast(GraphQLObjectType, self._schema.query_type)
         fields.update(query_type.fields)
+        query_type_kwargs = query_type.to_kwargs()
+        query_type_kwargs["fields"] = fields
 
-        self._schema.query_type = GraphQLObjectType(
-            name=query_type.name,
-            description=query_type.description,
-            fields=fields,
-        )
+        self._schema.query_type = GraphQLObjectType(**query_type_kwargs)
```

Comment:
Hmm, I'm not a fan of using `**kwargs`-style syntax unless absolutely necessary as it breaks type-checking. Is there a way to avoid this?

Reply:
I usually agree with you: I could just have added the `extensions=query_type.extensions` argument and have the tests pass.

However in this kind of scenario where we want a "copy with tweaks",  it is **very** easy to leave optional stuff behind (e.g., the code is currently missing `interfaces`, `is_type_of`, `extensions`, `ast_node` and `extension_ast_nodes`. And maybe more stuff in the future). 

Which is why I think `**kwargs` is a good choice here.

However I'll change If you are strongly against it

Reply:
Can we make use of [`copy`](https://docs.python.org/3/library/copy.html#module-copy) + [`GraphQLObjectType.__copy__`](https://github.com/graphql-python/graphql-core/blob/b9423b74ca9d22d009a1a4633b38b2bcd06c604b/src/graphql/type/definition.py#L728)?

Reply:
> Can we make use of [`copy`](https://docs.python.org/3/library/copy.html#module-copy) + [`GraphQLObjectType.__copy__`](https://github.com/graphql-python/graphql-core/blob/b9423b74ca9d22d009a1a4633b38b2bcd06c604b/src/graphql/type/definition.py#L728)?

This looks much better, thank you :)